File: Java/src/ejercicio2examen/Rectangulo.java
package ejercicio2examen;
/**
 * Clase Rectangulo del Ejercicio 2 del examen de Java. Crea un rectangulo con
 * dimensiones entre 1 y 10, y con el metodo toString se dibuja.
 * 
 * @author d18momoa
 *
 */
public class Rectangulo{
  private int ancho;
  private int alto;
  /**
   * Constructor que crea la clase rectangulo.
   * @param anc el ancho que recibe por parametro
   * @param alt el alto que recibe por parametro
   */
  public Rectangulo(int anc, int alt) {
    setAncho(anc);
    setAlto(alt);
  }
  /**
   * Constructor por defecto, crea un rectangulo con base 4 y altura 3
   */
  public Rectangulo() {
    setAncho(4);
    setAlto(3);
  }
  /**
   * Devuelve el valor actual del ancho del rectangulo
   * @return ancho del rectangulo
   */
  public int getAncho() {
    return ancho;
  }
  /**
   * Establece el valor del ancho, controla que estos valores esten comprendidos 
   * entre 1 y 10, en caso contrario lanzara la excepcion ArithmeticException.
   * @param ancho
   */
  protected void setAncho(int ancho) {
    if (ancho <= 0 || ancho > 10) {
      throw new ArithmeticException();
    } else {
      this.ancho = ancho;
    }
  }
  /**
   * Devuelve el valor actual del alto del rectangulo
   * @return alto del rectangulo
   */
  public int getAlto() {
    return alto;
  }
 /**
  * Establece el valor del alto, controla que estos valores esten comprendidos 
  * entre 1 y 10, en caso contrario lanzara la excepcion ArithmeticException.
  * @param ancho
  */
  protected void setAlto(int alto) {
    if (alto <= 0 || alto > 10) {
      throw new ArithmeticException();
    } else {
      this.alto = alto;
    }
  }
  /**
   * Devuelve en una cadena de texto el rectangulo pintado, de manera que al mostrarlo
   * en pantalla, muestre esa representacion grafica.
   */
  public String toString() {
    int i, espacios;
    String resultado = "";
    for (i = 0; i < this.ancho; i++) {
      resultado += "*";
    }
    resultado += "\n";
    for (i = 1; i < this.alto - 1; i++) {
      resultado += "*";
      for (espacios = 1; espacios < this.ancho - 1; espacios++) {
        resultado += " ";
      }
      resultado += "*\n";
    }
    for (i = 0; i < this.ancho; i++) {
      resultado += "*";
    }
    resultado += "\n";
    return resultado;
  }
  @Override
  public int hashCode() {
    final int prime = 31;
    int result = 1;
    result = prime * result + alto;
    result = prime * result + ancho;
    return result;
  }
  @Override
  public boolean equals(Object obj) {
    if (this == obj)
      return true;
    if (obj == null)
      return false;
    if (getClass() != obj.getClass())
      return false;
    Rectangulo other = (Rectangulo) obj;
    if (alto != other.alto)
      return false;
    if (ancho != other.ancho)
      return false;
    return true;
  }
}

File: EjerciciosEnPython/examen2Trim/Rectangulo.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
'''
Created on 5 abr. 2019
Clase Rectangulo creada para el examen del 2 Trimestre
@author: d18momoa
'''

class Rectangulo:
    """
    Constructor de la clase rectangulo, recibe por parametros dos enteros que conforman
    el ancho y el alto. Se verifican que los parametros sean correctos para poder asignarlos.
    """
    def __init__(self,ancho,alto):
        Rectangulo.__verifica_lado(ancho)
        Rectangulo.__verifica_lado(alto)
        self.__ancho = ancho
        self.__alto = alto
    """
    Creacion de las propiedades del ancho del rectangulo, implementa el setter 
    y el getter de este atributo.En el setter comprueba que su valor este 
    comprendido entre 1 y 10.
    """
    @property
    def ancho(self):
        return self.__ancho
    @ancho.setter
    def ancho(self, an):
        Rectangulo.__verifica_lado(an)
        self.__ancho = an
    """
    Creacion de las propiedades del alto del rectangulo, implementa el setter 
    y el getter de este atributo.En el setter comprueba que su valor este 
    comprendido entre 1 y 10.
    """
    @property
    def alto(self):
        return self.__alto
    @alto.setter
    def alto(self, al):
        Rectangulo.__verifica_lado(al)
        self.__alto = al
    """
    Devuelve una cadena de texto con el rectangulo pintado, de manera
    que cuando se muestre el objeto en pantalla, se vea su representacion
    grafica.
    """
    def __str__(self):
        cadena = ""
        cadena += "*"*self.ancho+"\n"
        if(self.alto == 2):
            cadena += "*"*self.ancho+"\n"
        elif(self.alto > 2):
            for i in range(0,self.alto-2):
                cadena+= "*"
                cadena+= " "*(self.ancho-2)
                cadena+= "*\n"
            cadena += "*"*self.ancho+"\n"
        return cadena
    """
    Metodo estatico de la clase cuadrado, que comprueba que el lado se encuentre
    entre 1 y 10 y que sea un entero. Si no es un entero lanzara un TypeError y si
    no comprende los valores entre 1 y 10, lanzara un ArithmeticError.
    Nos servira para verificar tanto el alto, como el ancho del Rectangulo.
    """
    @staticmethod
    def __verifica_lado(num):
        if not isinstance(num, int):  # lado no entero
            raise TypeError("Lado no entero", num) #Lanzo esta excepcion si el parametro introducido no es un entero.
        if (num <= 0 or num >10):
            raise ArithmeticError() #Lanzo esta excepcion que es similar al ArithmeticException de Java

"""
Main de la clase Rectangulo donde se llevan a cabo las pruebas.
"""
if __name__ == '__main__':
    try:
        r1 = Rectangulo(3,4)
        r2 = Rectangulo(5,10)
        #Descomentando la linea de abajo capturamos el ArithmeticError
        #r3 = Rectangulo(11,3)
        #Descomentando la linea de abajo capturamos el TypeError
        #r4 = Rectangulo(4.5,8)
        print("Pintamos el primer rectangulo cuyos datos son:")
        print("Ancho: "+str(r1.ancho))
        print("Alto: "+str(r1.alto))
        print(r1)
        print("--------------------------------------\n")
        print("Pintamos el segundo rectangulo cuyos datos son:")
        print("Ancho: "+str(r2.ancho))
        print("Alto: "+str(r2.alto))
        print(r2)
        print("--------------------------------------\n")
        print("Modifico el alto del r2 a 4")
        r2.alto = 4
        print("Muestro r2 modificado:")
        print(r2)
    except ArithmeticError:
        print("Error, los valores del rectangulo deben estar comprendidos entre 1 y 10")
    except TypeError:
        print("Error. Comprueba que el ancho y el alto sean valores enteros.")
    